src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -219,14 +224,14 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                     // Only add a comma after the first param, if any
                     builder.Append(',');
                 }
-                builder.Append(" error=\"invalid_token\", error_description=\"");
-                builder.Append(errorDescription);
+                builder.Append($" error=\"{eventContext.Error ?? "invalid_token"}\", error_description=\"");
+                builder.Append(eventContext.ErrorDescription ?? "An error occurred when validating the access token.");
                 builder.Append('\"');
 
                 Response.Headers.Append(HeaderNames.WWWAuthenticate, builder.ToString());
             }
 
-            return false;
+            return true;
```

Comment:
> The redirecting providers are all exclusive and would return true, but providers like Basic, Bearer, etc. can all contribute challenges to the same response in an additive fashion.

Why would non-interactive providers deserve a different treatment? There's fundamentally no difference between "redirecting providers" and the ones you mentioned.

Here, we could potentially end up with multiple `WWW-Authenticate` headers. While it's not forbidden by the HTTP spec (at least, AFAIK), it's not something most "REST clients" expect.

But if you think it's better, I'll revert this change.